pkg/events/events_test.go

```diff
@@ -165,3 +165,40 @@ func TestRecordEventWithCache(t *testing.T) {
 		t.Error("invalid Count")
 	}
 }
+
+func TestEventWithEmptyNamespaceReference(t *testing.T) {
+	clientMock := &k8sClientMock{}
+
+	recorder := events.NewEventRecorder(clientMock, newTestScheme(), events.EventsMap, events.EventRecorderOptions{
+		Version:   "1",
+		Cluster:   "cluster-1",
+		Component: "controller",
+		Namespace: "test",
+	})
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-pod",
+		},
+	}
+	err := recorder.RecordEvent(context.Background(), &events.Event{
+		Object:            pod,
+		RelatedObject:     nil,
+		ReportingInstance: "controller",
+		Name:              events.EventExampleEvent,
+	})
+	if err != nil {
```

Comment:
you should avoid using `conditionals` in test as much as possible, instead use asserts